cmd/clusterctl/client/yamlprocessor/processor.go

```diff
@@ -29,4 +29,12 @@ type Processor interface {
 	// Process processes the template blob of bytes and will return the final
 	// yaml with values retrieved from the values getter
 	Process([]byte, func(string) (string, error)) ([]byte, error)
+
+	// DefaultValues parses the template blob of bytes and provides a
+	// gets all value default if exists
+	DefaultValues([]byte) (map[string]string, error)
+
+	// RequiredFields parses the template blob of bytes and provides
+	// required field in this template
+	RequiredFields([]byte) ([]string, error)
```

Comment:
This is not ideal because we are asking all the yaml processor to implement two new methods, somehow related with `GetVariables` but separated

Reply:
the reason I didn't modify datatype is that it was being used by other files as type []string

Reply:
first time writing golang. How should i structure this ?

Reply:
this comment is now obsolete due to https://github.com/kubernetes-sigs/cluster-api/pull/4262#discussion_r589576031